Fabric.Authorization.API/Infrastructure/Middleware/AngularMiddleware.cs

```diff
@@ -38,18 +38,40 @@ public async Task Invoke(HttpContext context)
                 var discoveryServiceSettings = _appConfiguration.AccessControlSettings.DiscoveryServiceSettings;
                 _indexContent = _indexContent.Replace(discoveryServiceSettings.Token,
                     discoveryServiceSettings.Value);
-
-                File.WriteAllText(fullPath, _indexContent);
             }
 
-            await _next(context);
-            
-            if (context.Response.StatusCode == (int) HttpStatusCode.NotFound &&
-                !Path.HasExtension(context.Request.Path.Value) &&
-                context.Request.Path.Value.StartsWith(accessControlPath))
+            if (context.Request.Path == indexPath)
+            {
+                await WriteIndexResponse(context);
+            }
+            else
             {
-                context.Request.Path = indexPath;
                 await _next(context);
+                if (context.Response.StatusCode == (int)HttpStatusCode.NotFound &&
+                    !Path.HasExtension(context.Request.Path.Value) &&
+                    context.Request.Path.Value.StartsWith(accessControlPath))
+                {
+                    context.Request.Path = indexPath;
+                    await WriteIndexResponse(context);
+                }
+            }
+        }
+
+        private async Task WriteIndexResponse(HttpContext context)
+        {
+            using (var memoryStream = new MemoryStream(Encoding.ASCII.GetBytes(_indexContent)))
+            {
+                var originalResponse = context.Response.Body;
+                try
+                {
```

Comment:
Does the response body byte size need to be adjusted, or does it automatically get set correctly? I remember reading something about that online when I was working on this.